### PR TITLE
Improve exception catching in UCX communication

### LINKED ISSUE #7130: Catch exceptions on both send and recv in UCX comm
Following on from a review of #6996, there are a few other places in the UCX communicator backend where we should be catching exceptions on message transfer.

In addition, some of the try catch blocks can have their scope tightened (to avoid hiding other errors).

cc @pentschev, I will self-assign, but don't have privileges to do so.

### PR DESCRIPTION
**Pull request by pentschev**
Closes #7130

As was previously discussed in #6996, some of the exception catching could be improved for both read and write tasks which are now addressed by the changes contained here.

- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by pentschev** [comment]
cc @wence-

**Comment by wence-** [comment]
Lints fails are addressed by #7131

**Review by wence- — APPROVED** [review]
Very minor typo, but otherwise looks good, thanks!

File: distributed/comm/ucx.py [hunk 1]
```diff
@@ -352,14 +353,29 @@ async def read(self, deserializers=("cuda", "dask", "pickle", "error")):
...
-                frames,
-                deserialize=self.deserialize,
-                deserializers=deserializers,
-                allow_offload=self.allow_offload,
-            )
+            try:
+                for each_frame in recv_frames:
+                    await self.ep.recv(each_frame)
+            except BaseException as e:
+                # In addition to UCX exceptions, may be CancelledError or a another
```
[the remark this answers is not in the document]
Comment (pentschev) [reply distributed/comm/ucx.py]: Thanks Lawrence, I fixed it in https://github.com/dask/distributed/pull/7132/commits/e5d289fbe38b4ca242200d277557fdbea2b9c706 along with the original comment where I copy/pasted from. 🙂

**Comment by wence-** [comment]
I think if you merge trunk (or rebase) the linting errors will disappear

**Comment by quasiben** [comment]
The [test_shutdown_localcluster](https://github.com/dask/distributed/actions/runs/3229045964/jobs/5285895504#step:18:1305) failure is a known flaky tests:
https://github.com/dask/distributed/issues/6625#issuecomment-1211725774

The [test_sni](https://github.com/dask/distributed/actions/runs/3229045964/jobs/5285895625#step:18:1959) failure is also unrelated but I've not seen it before.  I'm going to restart CI but it would be good to keep an eye on it:

```
___________________________________ test_sni ___________________________________

loop = <tornado.platform.asyncio.AsyncIOMainLoop object at 0x7fa96cb75730>

    def test_sni(loop):
        port = open_port()
        with popen(["dask-scheduler", "--no-dashboard", f"--port={port}"] + tls_args) as s:
            with popen(
                [
                    "dask-worker",
                    "--no-dashboard",
                    "--scheduler-sni",
                    "localhost",
                    f"tls://127.0.0.1:{port}",
                ]
                + tls_args
            ) as w:
                with Client(
                    f"tls://127.0.0.1:{port}", loop=loop, security=tls_security()
                ) as c:
>                   wait_for_cores(c)

distributed/cli/tests/test_tls_cli.py:60: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

c = <Client: 'tls://10.213.3.122:50230' processes=0 threads=0, memory=0 B>
nthreads = 1

    def wait_for_cores(c, nthreads=1):
        start = time()
        while len(c.nthreads()) < 1:
            sleep(0.1)
>           assert time() < start + 10
E           assert 1665511027.440665 < (1665511017.4086459 + 10)
E            +  where 1665511027.440665 = time()

distributed/cli/tests/test_tls_cli.py:30: AssertionError
```

**Comment by quasiben** [comment]
All tests pass now -- merging in.

@wence- if you have thoughts on improving either of the flaky tests they would be extremely appreciated

**Pull request merged**